### PR TITLE
docs: add doc to event-management api

### PR DESCRIPTION
**Pull request by jbogarthyde**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] angular.io application / infrastructure changes
[ ] Other... Please describe:
```

## What is the current behavior?

Issue Number:  #21146

## What is the new behavior?
doc added

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
Closed previous PR, too old to sync. Needs new review.

### CODE REVIEW

File: packages/platform-browser/src/dom/events/hammer_gestures.ts [hunk 1]
```diff
@@ -12,6 +12,9 @@ import {DOCUMENT} from '../dom_tokens';
 
 import {EventManagerPlugin} from './event_manager';
 
+/**
+ * Supported HammerJS recognizer event names.
```
Comment (petebacondarwin) [inline packages/platform-browser/src/dom/events/hammer_gestures.ts]: Perhaps a link to what a HammerJS recognizer is would be helpful here?

File: packages/platform-browser/src/dom/events/hammer_gestures.ts [hunk 2]
```diff
@@ -51,8 +54,8 @@ const EVENT_NAMES = {
 };
 
 /**
- * A DI token that you can use to provide{@link HammerGestureConfig} to Angular. Use it to configure
- * Hammer gestures.
+ * DI token for providing {@link http://hammerjs.github.io/ HammerJS} support to Angular.
```
Comment (petebacondarwin) [inline packages/platform-browser/src/dom/events/hammer_gestures.ts]: It is actually preferable not to use `{@link ... }` for external links any more.
(I am trying to get rid of this inline jsdoc tag altogether)
It is cleaner (and works better with markdown syntax highlighting etc) if you just did:

```
 DI token for providing [HammerJS](http://hammerjs.github.io/) support to Angular.
```

File: packages/platform-browser/src/dom/events/hammer_gestures.ts [hunk 3]
```diff
@@ -51,8 +54,8 @@ const EVENT_NAMES = {
...
+ * @see  {@link HammerGestureConfig HammerGestureConfig}
```
Comment (petebacondarwin) [inline packages/platform-browser/src/dom/events/hammer_gestures.ts]: No need for `{@link ... }` here . Just putting the API item in backticks should be enough to turn it into a link:

```
@see `HammerGestureConfig`
```

File: packages/platform-browser/src/dom/events/hammer_gestures.ts [hunk 4]
```diff
@@ -81,8 +114,15 @@ export class HammerGestureConfig {
     inputTarget?: EventTarget;
   };
 
+  /**
+   * @description Creates a {@link http://hammerjs.github.io/api/#hammer.manager HammerJS Manager}
+   * and attaches it to a given HTML element.
+   * @param element The element that will recognize gestures.
```
Comment (petebacondarwin) [inline packages/platform-browser/src/dom/events/hammer_gestures.ts]: Does the element recognize gestures? Or is it just the source of the gesture?

File: packages/platform-browser/src/dom/events/key_events.ts [hunk 5]
```diff
@@ -23,14 +30,32 @@ const MODIFIER_KEY_GETTERS: {[key: string]: (event: KeyboardEvent) => boolean} =
 
 /**
  * @experimental
+ * @description An injectable service that provides support for handling of key events in Angular.
  */
 @Injectable()
 export class KeyEventsPlugin extends EventManagerPlugin {
+  /**
+   * Initializes an instance of the key-event management service.
```
Comment (petebacondarwin) [inline packages/platform-browser/src/dom/events/key_events.ts]: Is "management service" accurate? Isn't this initializing a key-event plugin?

File: packages/platform-browser/src/dom/events/key_events.ts [hunk 6]
```diff
@@ -23,14 +30,32 @@ const MODIFIER_KEY_GETTERS: {[key: string]: (event: KeyboardEvent) => boolean} =
...
+    * Reports whether a named key event is supported.
+    * @param eventName The event name to query.
+    * @return True if the named key event is supported.
+   */
   supports(eventName: string): boolean { return KeyEventsPlugin.parseEventName(eventName) != null; }
 
-  addEventListener(element: HTMLElement, eventName: string, handler: Function): Function {
+   /**
+    * Registers a handler for a specific element and key event.
+    * @param element The HTML element to receive event notifications.
```
Comment (petebacondarwin) [inline packages/platform-browser/src/dom/events/key_events.ts]: Does the element receive or trigger events?

File: packages/platform-browser/src/dom/events/key_events.ts [hunk 7]
```diff
@@ -72,7 +102,11 @@ export class KeyEventsPlugin extends EventManagerPlugin {
     return result;
   }
 
-  static getEventFullKey(event: KeyboardEvent): string {
+   /**
+    * Constructs an event name string from events that combine multiple key actions.
+    * @param event The keyboard-event object.
+    */
```
Comment (petebacondarwin) [inline packages/platform-browser/src/dom/events/key_events.ts]: Missing `@returns`?

File: packages/platform-browser/src/dom/events/key_events.ts [hunk 8]
```diff
@@ -93,6 +127,12 @@ export class KeyEventsPlugin extends EventManagerPlugin {
     return fullKey;
   }
 
+  /**
+   * Event-handling callback for a key event.
+   * @param fullKey The event name that combines all simultaneous keystrokes.
+   * @param handler The function that responds to the key event.
+   * @param zone The zone in which the event occurred.
```
Comment (petebacondarwin) [inline packages/platform-browser/src/dom/events/key_events.ts]: Missing `@returns`?

**Review by petebacondarwin — CHANGES_REQUESTED** [review]
I would try to remove all the `@description` and `{@link}` usage.
I noted a few content issues, that you may want to consider.

File: packages/platform-browser/src/dom/events/event_manager.ts [hunk 9]
```diff
@@ -13,52 +13,57 @@ import {getDOM} from '../dom_adapter';
 /**
  * The injection token for the event-manager plug-in service.
  */
-export const EVENT_MANAGER_PLUGINS =
+ export const EVENT_MANAGER_PLUGINS =
```
Comment (petebacondarwin) [inline packages/platform-browser/src/dom/events/event_manager.ts]: pretty sure this should not be indented.

File: packages/platform-browser/src/dom/events/event_manager.ts [hunk 10]
```diff
@@ -13,52 +13,57 @@ import {getDOM} from '../dom_adapter';
...
     new InjectionToken<EventManagerPlugin[]>('EventManagerPlugins');
 
+@Injectable()
```
Comment (petebacondarwin) [inline packages/platform-browser/src/dom/events/event_manager.ts]: And this.

File: packages/platform-browser/src/dom/events/event_manager.ts [hunk 11]
```diff
@@ -13,52 +13,57 @@ import {getDOM} from '../dom_adapter';
...
+ export class EventManager {
   private _plugins: EventManagerPlugin[];
   private _eventNameToPlugin = new Map<string, EventManagerPlugin>();
  /**
-  * Initializes an instance of the event-manager service.
-  * @param plugins The event-management plug-ins to provide.
-  * @param _zone The compilation zone.
+  * 
+  * 
+  *
```
Comment (petebacondarwin) [inline packages/platform-browser/src/dom/events/event_manager.ts]: Why add this empty comment?

File: packages/platform-browser/src/dom/events/event_manager.ts [hunk 12]
```diff
@@ -13,52 +13,57 @@ import {getDOM} from '../dom_adapter';
...
+   * @returns A callback function that can be used to remove the handler.
+   */
   addGlobalEventListener(target: string, eventName: string, handler: Function): Function {
     const plugin = this._findPluginFor(eventName);
     return plugin.addGlobalEventListener(target, eventName, handler);
   }
- /**
-  * Retrieves the compilation zone in which event listeners are registered.
-  */
+  /**
```
Comment (petebacondarwin) [inline packages/platform-browser/src/dom/events/event_manager.ts]: Should there be an empty line before this comment?

File: packages/platform-browser/src/dom/events/hammer_gestures.ts (hunk 1 above)
Comment (jbogarthyde) [reply packages/platform-browser/src/dom/events/hammer_gestures.ts]: They don't explain what it is or define it in their doc. Perhaps @vicb can define it?

File: packages/platform-browser/src/dom/events/event_manager.ts [hunk 13]
```diff
@@ -11,34 +11,54 @@ import {Inject, Injectable, InjectionToken, NgZone} from '@angular/core';
...
 /**
- *
+ * The injection token for the event-manager plug-in service.
  */
 export const EVENT_MANAGER_PLUGINS =
     new InjectionToken<EventManagerPlugin[]>('EventManagerPlugins');
 
 /**
- *
+ * @description An injectable service that provides event management in the browser.
```
[the remark this answers is not in the document]
Comment (jbogarthyde) [reply packages/platform-browser/src/dom/events/event_manager.ts]: Added a minimal statement - @vicb, can you provide more context?

File: packages/platform-browser/src/dom/events/hammer_gestures.ts (hunk 4 above)
Comment (jbogarthyde) [reply packages/platform-browser/src/dom/events/hammer_gestures.ts]: I don't know. @vicb ?

File: packages/platform-browser/src/dom/events/key_events.ts (hunk 5 above)
Comment (jbogarthyde) [reply packages/platform-browser/src/dom/events/key_events.ts]: I think you're right, but I also think this file is not getting into the doc at all.

File: packages/platform-browser/src/dom/events/key_events.ts (hunk 6 above)
Comment (jbogarthyde) [reply packages/platform-browser/src/dom/events/key_events.ts]: I don't know either, but this is not getting into the doc. Removing comments.

File: packages/platform-browser/src/dom/events/hammer_gestures.ts [hunk 14]
```diff
@@ -64,14 +67,44 @@ export interface HammerInstance {
...
 
 /**
+ * An injectable [HammerJS Manager](http://hammerjs.github.io/api/#hammer.manager)
+ * for gesture recognition. Configures specific event recognition.
  * @experimental
  */
 @Injectable()
 export class HammerGestureConfig {
+  /**
+   * @description A set of supported event names for gestures to be used in Angular.
```
Comment (IgorMinar) [inline packages/platform-browser/src/dom/events/hammer_gestures.ts]: the @description tag is not necessary in this case

File: packages/platform-browser/src/dom/events/event_manager.ts [hunk 15]
```diff
@@ -11,34 +11,57 @@ import {Inject, Injectable, InjectionToken, NgZone} from '@angular/core';
...
+  /**
+   * Initializes an instance of the event-manager service.
+   */
   constructor(@Inject(EVENT_MANAGER_PLUGINS) plugins: EventManagerPlugin[], private _zone: NgZone) {
     plugins.forEach(p => p.manager = this);
     this._plugins = plugins.slice().reverse();
   }
 
+  /**
+   * Registers a handler for a specific element and event.
```
Comment (IgorMinar) [inline packages/platform-browser/src/dom/events/event_manager.ts]: could we add a blank line that separates the description from params? the blank line makes the doc block easier to read in the source form and doesn't affect the presentation on angular.io. thanks!

File: packages/platform-browser/src/dom/events/hammer_gestures.ts [hunk 16]
```diff
@@ -81,8 +114,15 @@ export class HammerGestureConfig {
     inputTarget?: EventTarget;
   };
 
+  /**
+   * Creates a [HammerJS Manager](http://hammerjs.github.io/api/#hammer.manager)
+   * and attaches it to a given HTML element.
+   * @param element The element that will recognize gestures.
+   * @returns A HammerJS event-manager object.
+   */
+
```
Comment (IgorMinar) [inline packages/platform-browser/src/dom/events/hammer_gestures.ts]: please remove this blank line. by convention the docs block should be attached to the prop declaration.

**Review by IgorMinar — APPROVED** [review]
lgtm except for a few small nits

File: packages/platform-browser/src/dom/events/hammer_gestures.ts (hunk 16 above)
Comment (jbogarthyde) [reply packages/platform-browser/src/dom/events/hammer_gestures.ts]: corrections made

**Review by petebacondarwin — APPROVED** [review]

**Pull request closed without merging**